File: webapp/src/app/model/league.ts
export interface League {
  id: number;
  name: string;
  logoUrl: string;
  country: Country;
  isCurrent?: boolean;
  seasonStart: number;
  seasonEnd: number;
}

export interface SimpleLeague {
  id: number;
  name: string;
  country: string;
  logoUrl: string;
  flagUrl: string;
}

export interface Country {
  id?: string;
  name: string;
  flagUrl?: string;
}

export interface Leagues {
  userLeagues: League[];
  availableLeagues: League[];
}

export interface LeagueId {
  leagueId: number;
}

File: main/src/main/java/com/smartbet/demo/league/domain/League.java
package com.smartbet.demo.league.domain;

import com.smartbet.demo.country.domain.Country;
import lombok.Data;

import java.time.LocalDate;

@Data
public class League {
    private final int id;
    private final String name;
    private final String logoUrl;
    private final Country country;
    private final boolean isCurrent;
    private final LocalDate seasonStart;
    private final LocalDate seasonEnd;
}
